connector/session.py

```diff
@@ -202,6 +202,11 @@ def rollback(self):
 
     def close(self):
         """ Close the cursor """
+        try:
+            # Try to recompute fields before closing the cursor
+            self.env['ir.model'].recompute()
+        except:
+            self.rollback()
```

Comment:
> This means we would remove the todo instead of calculating them. It should come to the same result since they are stored fields, and is probably cleaner than trying to calculate them.

Yes and calculating them seems useless if anyway the transaction is broken.

> But shouldn't we keep this cleaning in ConnectorSession.close() in case we get the error from somewhere else ?
> Or maybe we could modify ConnectorSessionHandler.session() by using clear_upon_failure this way ?

I would prefer if we could first prove that it could happen somewhere else and then fix this other place.

Also `ConnectorSession` is quite redundant with Odoo's `Environment` and in the future we could start to remove `ConnectorSession` in favor of `openerp.Environment` so we should try not to add behavioural changes that we wouldn't know how to replace.